SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -4209,3 +4211,111 @@ void CSolver::BasicLoadRestart(CGeometry *geometry, const CConfig *config, const
+  const int maxBasisDim          = config->GetMax_BasisDim();
+  int dim = int(nPointDomain * nVar);
+  bool incremental = false;
+
+  if (!u_basis_generator) {
+    
+    /*--- Define SVD basis generator ---*/
+    int timesteps = (int)(nTimeIter - TimeIter);
+    CAROM::Options svd_options = CAROM::Options(dim, timesteps, -1,
+                                                false, true).setMaxBasisDimension(int(maxBasisDim));
+    
+    if (config->GetKind_PODBasis() == POD_KIND::STATIC) {
+      if (rank == MASTER_NODE) std::cout << "Creating static basis generator." << std::endl;
+      
+      if (unsteady) {
+        if (rank == MASTER_NODE) std::cout << "Incremental basis generator recommended for unsteady simulations." << std::endl;
+      }
+    }
+    else {
+      if (rank == MASTER_NODE) std::cout << "Creating incremental basis generator." << std::endl;
+      
+      svd_options.setIncrementalSVD(1.0e-3, config->GetDelta_UnstTime(),
+                                    1.0e-2, config->GetDelta_UnstTime()*nTimeIter, true).setDebugMode(false);
+      incremental = true;
+    }
+    
+    u_basis_generator.reset(new CAROM::BasisGenerator(
+      svd_options, incremental,
+      filename));
+    
+    // Print nodes for each rank for now
+    std::cout << "nPointDomain: " << nPointDomain << " for rank: " << rank << std::endl;
+    
+    // Save mesh ordering
+    std::ofstream f;
+    f.open(filename + "_mesh_" + to_string(rank) + ".csv");
+      for (unsigned long iPoint = 0; iPoint < nPointDomain; iPoint++) {
+        unsigned long globalPoint = geometry->nodes->GetGlobalIndex(iPoint);
+        auto Coord = geometry->nodes->GetCoord(iPoint);
+        f << Coord[0] << ", " << Coord[1] << ", " << globalPoint << "\n";
+      }
+    
+    f.close();
+  }
+
+  if (unsteady) {
+    double* u = new double[nPointDomain*nVar];
+    for (unsigned long iPoint = 0; iPoint < nPointDomain; iPoint++) {
+      for (unsigned short iVar = 0; iVar < nVar; iVar++) {
+        unsigned long total_index = iPoint*nVar + iVar;
+        u[total_index] = base_nodes->GetSolution(iPoint,iVar);
+      }
+    }
+    
+    // give solution and time steps to libROM:
+    double dt = config->GetDelta_UnstTime();
+    double t =  config->GetCurrent_UnstTime();
+    u_basis_generator->takeSample(u, t, dt);
+    delete[] u;
+  }
+   
+  /*--- End collection of data and save POD ---*/
+  
+  if (converged) {
+  
+    if (!unsteady) {
+       double* u = new double[nPointDomain*nVar];
+       for (unsigned long iPoint = 0; iPoint < nPointDomain; iPoint++) {
+         for (unsigned short iVar = 0; iVar < nVar; iVar++) {
+             unsigned long total_index = iPoint*nVar + iVar;
+             u[total_index] = base_nodes->GetSolution(iPoint, iVar);
+         }
+       }
+ 
+       // dt is different for each node, so just use a placeholder dt
+       double dt = base_nodes->GetDelta_Time(0);
+       double t = dt*TimeIter;
+       u_basis_generator->takeSample(u, t, dt);
```

Comment:
```suggestion
       u_basis_generator->takeSample(base_nodes->GetSolution().data(), t, dt);
```

This should work but it will not compile with AD, hence the modified #ifdef.